threedigrid_builder/tests/test_lines_1d2d.py

```diff
@@ -179,6 +179,18 @@ def test_assign_2d_node(side_2d_coordinates, expected_2d_node_id, cell_tree):
     assert_array_equal(lines.line_coords, np.nan)  # is cleared
 
 
+def test_remove_line_unassigned_nodes():
+    nodes = Nodes(id=[1, 2])
+    lines = Lines1D2D(
+        id=[1, 2], line=[(-9999, 1), (1, 2)], line_coords=[(10, 0, 0, 0), (5, 2, 3, 8)]
+    )
+    lines.remove_unassigned(nodes)
+    assert lines.id == [2]
```

Comment:
for comparing numpy arrays, you should use `np.testing.assert_equal` or (for floats) `np.testing.assert_almost_equal`.